### PR TITLE
Feature/schedule

### PR DESCRIPTION
**Pull request by egfiskeseth**

### CODE REVIEW

File: pages/components/ScheduleEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const EventHeader = props => {
+  return (
+    <div>
+      <h3>{props.eventHeader}</h3>
+      <p>{props.eventTime}</p>
+    </div>
+  );
+};
+
+const Talk = props => {
```
Comment (Maritt) [inline pages/components/ScheduleEntry.tsx]: should talk perhaps be a separate component?

File: pages/components/ScheduleEntry.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          type="Short talk"
+          language="English"
+          eventTime={props.eventTime}
+        />
+      </div>
+    </div>
+  );
+}
+
+const EventHeader = props => {
```
Comment (Maritt) [inline pages/components/ScheduleEntry.tsx]: Should EventHeader be a separate component?

File: styling/scheduleStyles.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
+$break-small: 650px;
+$break-large: 1200px;
+
+.session {
+  background-color: #c9e2e0;
```
Comment (Maritt) [inline styling/scheduleStyles.scss]: color move into variables file

File: pages/components/Day.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+  render() {
+    return (
+      <div>
+        {this.state.clicked && (
+          <div>
+            <ShowScheduleButton
+              showSchedule={this.showSchedule}
+              day={this.props.currDay}
+              timeStart={this.props.slots[0].timeStart}
```
Comment (Maritt) [inline pages/components/Day.tsx]: should possibly not hardcode the index of slot?

File: pages/components/Day.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    return (
+      <ShowDailySchedule
+        day={props.day}
+        slots={props.slots}
+        showSchedule={props.showSchedule}
+        timeStart={props.timeStart}
+      />
+    );
+  } else {
+    if (props.day == "Fredag") {
```
Comment (Maritt) [inline pages/components/Day.tsx]: the only difference here seems to be the day. Should probably be refactored and use the day string from props.day

File: pages/components/ReadData.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import "../models/data.json";
+
+const ReadData = async () => {
```
Comment (Maritt) [inline pages/components/ReadData.tsx]: is this in use?

File: pages/components/ScheduleEntry.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  render() {
+
+    if (this.props.slot.rooms) {
+      return (
+        <div className="content">
+          <EventHeader timeStart={this.props.slot.timeStart} timeEnd={this.props.slot.timeEnd} type={this.props.slot.type} />
+          <div className="talk-slot">
+
+            {this.props.slot.rooms.map(room =>
+              <Talk title={room.talks[0].title}
```
Comment (Maritt) [inline pages/components/ScheduleEntry.tsx]: should not hardcode indices

File: pages/components/ScheduleEntry.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                type={room.talks[0].type}
+                language={room.talks[0].language}
+                difficulty={room.talks[0].difficulty} />
+            )}
+          </div>
+        </div>
+      );
+    } else {
+      return (<div className="content">
+        <EventHeader timeStart={this.props.slot.timeStart} timeEnd={this.props.slot.timeEnd} type={this.props.slot.type} />
```
Comment (Maritt) [inline pages/components/ScheduleEntry.tsx]: possibly add a comment of why this happens :)

File: pages/components/ShowDailySchedule.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import ScheduleEntry from "./ScheduleEntry";
+import EventHeader from "./EventHeader";
+import { loadGetInitialProps } from "next-server/dist/lib/utils";
+
+const ShowDailySchedule = props => {
+  if (props.day == "Fredag") {
```
Comment (Maritt) [inline pages/components/ShowDailySchedule.tsx]: this code is very similar to Day.tsx, we should reuse

File: pages/components/Talk.tsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
+const Talk = props => {
+  return (
+
+
+    <div className="talk-container">
+
+      <h3 id="title">
+        <a href="./talksAndSpeakers">{props.title}</a>
```
Comment (Maritt) [inline pages/components/Talk.tsx:8]: does this link to the actual speaker/talk or just to the correct page?

File: styling/scheduleStyles.scss [hunk 11]
```diff
@@ -0,0 +1,91 @@
+@import "variables";
+@import url("https://fonts.googleapis.com/css?family=Raleway&display=swap");
+
+.talk-slot {
+  @media screen and (min-width: $break-large) {
```
Comment (Maritt) [inline styling/scheduleStyles.scss]: should not be necessary with two mediaqueries here.

File: pages/components/Talk.tsx (hunk 10 above)
Comment (egfiskeseth) [reply pages/components/Talk.tsx:8]: Currently just the speaker page, have not implemented this yet

File: pages/components/Day.tsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+      this.setState({
+        clicked: true
+      });
+    }
+  };
+
+  render() {
+    return (
+      <div>
+        {this.state.clicked && (
```
Comment (Maritt) [inline pages/components/Day.tsx]: this and the other seems to be the excact same? is it necessary to create two buttons when you set clicked prop in the component?

File: pages/components/ScheduleEntry.tsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  constructor(props) {
+    super(props)
+  }
+
+  render() {
+
+    if (this.props.slot.rooms) {
+      return (
+        <div className="content">
+          <EventHeader timeStart={this.props.slot.timeStart} timeEnd={this.props.slot.timeEnd} type={this.props.slot.type} />
```
Comment (Maritt) [inline pages/components/ScheduleEntry.tsx]: could eventheader be moved out of this if to minimize coderepetition?

**Review by Maritt — APPROVED** [review]

**Pull request merged**